Chore: Upgrade grafana-app-sdk v0.46.0

gillesdemey opened the pull request:
**What is this feature?**

This PR bumps `grafana-app-sdk` to [v0.46.0](https://github.com/grafana/grafana-app-sdk/releases/tag/v0.46.0)

* Makes `spec` et al. non-optional as per https://github.com/grafana/grafana-app-sdk/pull/1000
* Passing a custom `ErrorHandler` to `InformerConfig` is now deprecated, these were all using the default templated behaviour. (https://github.com/grafana/grafana-app-sdk/pull/895/files#r2359331454)

**Special notes for your reviewer:**

This PR's diff is rather large but should be mostly digestible if we ignore `.sum` and `.mod` file updates.

gillesdemey committed:
bump codegen to grafana-app-sdk v0.46.0

gillesdemey committed:
go fmt ✨

gillesdemey commented:
Gotcha – I'm assuming the way to configure one has changed though since the  `InformerConfig{}` interface was updated to omit `ErrorHandler`? I haven't really found any apps internally that do anything meaningful so I've opted not to refactor those.

radiohead commented on public/app/features/playlist/utils.ts:
```diff
@@ -71,6 +71,8 @@ export async function loadDashboards(items: PlaylistItemUI[]): Promise<PlaylistI
 export function getDefaultPlaylist(): Playlist {
   return {
+    kind: 'Playlist',
+    apiVersion: '',
```
Are these hand-rolled? I think the version should use the preferred version (though not sure if there's already a way to get that dynamically from the API):
```suggestion
    apiVersion: 'v0alpha1',
```

radiohead reviewed (commented):
Just a quick question about the missing `apiVersion` in the Playlist TS helper, otherwise looks reasonable to me.

gillesdemey committed:
make update-workspace

gillesdemey committed:
Fix PlaylistForm test

gillesdemey committed:
use appropriate apiVersions for kubernetes entities

gillesdemey committed:
update iam app to use the newer interfaces

gillesdemey committed:
typecheck ✅

gillesdemey committed:
revert custom ErrorHandlers

to make sure we aren't introducing any breaking changes in logging I've
opted to restore the custom error handlers where the default app-sdk
logger wasn't being used
